File: LAB6/zadanie 6.1/lab_work/task2/task2.go
/**
 * @file task2.go
 * @brief Генерация матрицы по шаблону
 * 
 * Программа создает квадратную матрицу по формуле:
 * a[i][j] = 100 + 5*(i + j)
 * 
 * @version 1.0
 * @author Student Name
 */

package main

import "fmt"

func main() {
	var N int
	fmt.Print("Введите количество строк матрицы: ")
	fmt.Scan(&N)

	// Создание матрицы
	matrix := make([][]int, N)
	for i := range matrix {
		matrix[i] = make([]int, N)
		for j := range matrix[i] {
			matrix[i][j] = 100 + 5*(i+j)
		}
	}

	// Вывод матрицы
	fmt.Println("\nСгенерированная матрица:")
	for _, row := range matrix {
		for _, val := range row {
			fmt.Printf("%4d", val)
		}
		fmt.Println()
	}
}

File: LAB6/zadanie 6.1/lab_work/task2/task2.cpp
/**
 * @file task2.cpp
 * @brief Генерация матрицы по шаблону
 * 
 * Программа создает квадратную матрицу по формуле:
 * a[i][j] = 100 + 5*(i + j)
 * 
 * @version 1.0
 * @author Student Name
 */

#include <iostream>
#include <iomanip>
#include <vector>
using namespace std;

int main() {
    int N;
    cout << "Введите количество строк матрицы: ";
    cin >> N;

    // Создание матрицы по формуле
    vector<vector<int>> matrix(N, vector<int>(N));
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            matrix[i][j] = 100 + 5 * (i + j);
        }
    }

    // Вывод матрицы
    cout << "\nСгенерированная матрица:\n";
    for (const auto& row : matrix) {
        for (int val : row) {
            cout << setw(4) << val;
        }
        cout << endl;
    }
    
    return 0;
}
